dot/network/host.go

```diff
@@ -82,7 +82,11 @@ type host struct {
 
 func newHost(ctx context.Context, cfg *Config) (*host, error) {
 	// create multiaddress (without p2p identity)
-	addr, err := ma.NewMultiaddr(fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", cfg.Port))
+	addrString := fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", cfg.Port)
+	if cfg.ListenAddress != "" {
+		addrString = cfg.ListenAddress
+	}
```

Comment:
In this case we need to parse the port from the address supplied and store it in `cfg.Port`. `cfg.Port` is used to populate `externalAddr` further down.

Reply:
Why wouldn't we continue to use the value in `cfg.Port` for building the `externalAddr`?  Since listen address and external address may be different, should we assume that external address uses the same port as listen address when listen address cfg value has been set?